### ISSUE TITLE
Torch triangular solve breaks for high dimensional array on GPU

### ISSUE THREAD

**Issue by acse-gpm22** [dpelacani/AWLoss#18]
As discussed in this [PyTorch issue](https://github.com/pytorch/pytorch/issues/97211), `torch.linalg.solve_triangular` breaks on GPU for high dimensional data (dim > 524281), which is caused by the usage of `torch.distributions.multivariate_normal.MultivariateNormal.log_prob` [here](https://github.com/dpelacani/AWLoss/blob/f2be0223aba276607e2a7f6e37a20a3558753a07/awloss/awloss.py#L235).

It follows a simple code snippet that reproduces the error:

    import torch
    from awloss import AWLoss

    awloss = AWLoss()
    x = torch.rand([1, 3, 512, 512], device="cuda")
    y = torch.rand([1, 3, 512, 512], device="cuda")

    awloss(x, y)

Even if a user-specified penalty function is used, the [delta calculation](https://github.com/dpelacani/AWLoss/blob/f2be0223aba276607e2a7f6e37a20a3558753a07/awloss/awloss.py#L404) will still break.
